karton/core/backend.py

```diff
@@ -51,6 +65,33 @@ def __init__(self, config, identity: Optional[str] = None) -> None:
             secret_key=config["minio"]["secret_key"],
             secure=config.getboolean("minio", "secure", fallback=True),
         )
+        self.max_retries = self.config.getint("minio", "max_retries", 3)
+        self.retry_backoff = self.config.getint("minio", "retry_backoff", 5)
+
+    def do_retry(self, operation: Callable[[], T]) -> T:
+        """
+        Run HTTP operation and retry in case of errors
+        which may arise from temporary networking problems,
+        especially during download/upload operations.
+
+        minio-py handles only S3 errors and HTTP 502-504
+        (after completed response), so we need to retry whole
+        operation on our own when it fails in the middle of
+        transmission
+        """
+        tries_left = self.max_retries
+        while True:
+            try:
+                return operation()
+            except (ProtocolError, TimeoutError, IncompleteRead) as e:
+                if not tries_left:
+                    raise
+                tries_left -= 1
+                warnings.warn(
+                    f"Operation failed with {str(e)}. "
+                    f"Retrying after {self.retry_backoff} seconds..."
+                )
+                time.sleep(self.retry_backoff)
```

Comment:
should we maybe stay within the convention and use `backoff_factor` and a formula to calculate the timeout? `{backoff factor} * (2 ** ({number of total retries} - 1))`